spec/timerage/time_interval_spec.rb

```diff
@@ -187,49 +193,33 @@
   end
 
   context "exclusive 0 length interval" do
-    subject(:interval) { described_class.new(now...now) }
+    subject(:interval) { described_class.from_range(now...now) }
     specify { expect{ |b| subject.step(1, &b) }.not_to yield_control }
   end
 
   context "inclusive 0 length interval" do
-    subject(:interval) { described_class.new(now..now) }
+    subject(:interval) { described_class.from_range(now..now) }
     specify { expect{ |b| subject.step(1, &b) }.to yield_control.once }
   end
 
   context "includes leap day" do
-    subject(:interval) { described_class.new(before_leap_day..after_leap_day) }
+    subject(:interval) { described_class.new(before_leap_day, after_leap_day) }
     specify { expect{ |b| subject.step(1.day, &b) }.to yield_control.exactly(3).times }
     specify { expect{ |b| subject.step(86_400, &b) }.to yield_control.exactly(3).times }
   end
 
   context "transition into dst with explicit time zone" do
-    subject(:interval) { described_class.new(before_dst..after_dst) }
+    subject(:interval) { described_class.new(before_dst, after_dst) }
     specify { expect{ |b| subject.step(1.hour, &b) }.to yield_control.exactly(2).times }
     specify { expect{ |b| subject.step(3_600, &b) }.to yield_control.exactly(2).times }
   end
 
   context "transition into dst without explicit time zone" do
-    subject(:interval) { described_class.new(before_dst..(before_dst + 1.hour)) }
+    subject(:interval) { described_class.new(before_dst, (before_dst + 1.hour)) }
     specify { expect{ |b| subject.step(1.hour, &b) }.to yield_control.exactly(2).times }
     specify { expect{ |b| subject.step(3_600, &b) }.to yield_control.exactly(2).times }
   end
 
-  context "When given date range" do
-    it "returns date objects when stepping through the range with a day duration" do
-      date_range = Date.new(2020, 1, 1)..Date.new(2020,1,3)
-      described_class.new(date_range).step(1.day) do |date|
-        expect(date).to be_kind_of Date
-      end
-    end
-
-    it "returns Time objects when stepping through the range with an hour duration" do
-      date_range = Date.new(2020, 1, 1)..Date.new(2020,1,2)
-      described_class.new(date_range).step(6.hours) do |date|
-        expect(date).to be_kind_of Time
-      end
-    end
-  end
```

Comment:
Do we not need these specs anymore? I think I added these because the previous behaviour was consistent.

Reply:
good point